tools/floodgate/js/floodgate.js

```diff
@@ -12,15 +22,113 @@ import {
+      loadingON(`Copying ${srcPath} to pink folder`);
+      let copySuccess = false;
+      if (urlInfo?.doc?.fg?.sp?.status !== 200) {
+        const destinationFolder = `${srcPath.substring(0, srcPath.lastIndexOf('/'))}`;
+        copySuccess = await copyFile(srcPath, destinationFolder, undefined, true);
+        updateAndDisplayCopyStatus(copySuccess, srcPath);
+      } else {
+        // Get the source file
+        const file = await getFile(urlInfo.doc);
+        if (file) {
+          const destination = urlInfo?.doc?.filePath;
+          if (destination) {
+            // Save the file in the floodgate destination location
+            const saveStatus = await saveFile(file, destination, true);
+            if (saveStatus.success) {
+              copySuccess = true;
+            }
+          }
+        }
+        updateAndDisplayCopyStatus(copySuccess, srcPath);
+      }
+      status.success = copySuccess;
+      status.srcPath = srcPath;
+      status.dstPath = srcPath;
+      status.url = urlInfo.doc.url;
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.log(`Error occurred when trying to copy files to floodgated content folder ${error.message}`);
+    }
+    return status;
+  }
+
+  hideButtons(ACTION_BUTTON_IDS);
+  const startCopy = new Date();
+  const copyStatuses = await Promise.all(
+    [...projectDetail.urls].map((valueArray) => copyFilesToFloodgateTree(valueArray[1])),
+  );
+  const endCopy = new Date();
+
+  loadingON('Previewing for copied files... ');
+  const previewStatuses = await Promise.all(
+    copyStatuses
+      .filter((status) => status.success)
+      .map((status) => simulatePreview(stripExtension(status.dstPath), 1, true)),
+  );
+  loadingON('Completed Preview for copied files... ');
+
+  const failedCopies = copyStatuses.filter((status) => !status.success)
+    .map((status) => status.srcPath || 'Path Info Not available');
+  const failedPreviews = previewStatuses.filter((status) => !status.success)
+    .map((status) => status.path);
+
+  const excelValues = [['COPY', startCopy, endCopy, failedCopies.join('\n')]];
+  await updateExcelTable(project.excelPath, 'STATUS', excelValues);
+  loadingON('Project excel file updated with copy status... ');
+  showButtons(ACTION_BUTTON_IDS);
+
+  if (failedCopies.length > 0 || failedPreviews.length > 0) {
+    let failureMessage = failedCopies.length > 0 ? `Failed to copy ${failedCopies} to floodgate content folder. Check project excel sheet for additional information\n` : '';
+    failureMessage += failedPreviews.length > 0 ? `Failed to preview ${failedPreviews}. Kindly manually preview these files.` : '';
+    loadingON(failureMessage);
+  } else {
+    loadingOFF();
+    await refreshPage();
+  }
+}
+
 function setListeners() {
   document.querySelector('#reloadProject button').addEventListener('click', reloadProject);
+  document.querySelector('#copyFiles button').addEventListener('click', floodgateContent);
```

Comment:
To avoid globals
```suggestion
function setListeners(project, projectDetails) {
  document.querySelector('#reloadProject button').addEventListener('click', reloadProject);
  document.querySelector('#copyFiles button').addEventListener('click', () => floodgateContent(project, projectDetails));
```